src/main/java/org/craftercms/studio/impl/v2/service/security/internal/EncryptionServiceInternalImpl.java

```diff
@@ -19,51 +19,90 @@
+import org.craftercms.studio.api.v2.service.audit.internal.AuditServiceInternal;
 import org.craftercms.studio.api.v2.service.security.internal.EncryptionServiceInternal;
+import org.craftercms.studio.api.v2.utils.StudioConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import java.beans.ConstructorProperties;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.craftercms.studio.api.v2.dal.AuditLogConstants.OPERATION_CREATE;
+import static org.craftercms.studio.api.v2.dal.AuditLogConstants.TARGET_TYPE_ENCRYPTION_TOKEN;
+import static org.craftercms.studio.api.v2.dal.QueryParameterNames.SITE_ID;
+import static org.craftercms.studio.api.v2.utils.StudioConfiguration.CONFIGURATION_GLOBAL_SYSTEM_SITE;
 
 /**
  * @author joseross
  */
 public class EncryptionServiceInternalImpl implements EncryptionServiceInternal {
 
-    protected long delay;
+    private final static Logger logger = LoggerFactory.getLogger(EncryptionServiceInternalImpl.class);
 
-    protected int maxLength;
+    private final StudioConfiguration studioConfiguration;
+    private final AuditServiceInternal auditServiceInternal;
+    private final TextEncryptor textEncryptor;
+    private final SiteFeedMapper siteFeedMapper;
+    private final int maxLength;
+    private final long delay;
 
-    protected TextEncryptor textEncryptor;
+    @ConstructorProperties({"studioConfiguration", "auditServiceInternal", "textEncryptor", "siteFeedMapper",
+            "maxLength", "delay"})
+    public EncryptionServiceInternalImpl(StudioConfiguration studioConfiguration, AuditServiceInternal auditServiceInternal,
+                                         TextEncryptor textEncryptor, SiteFeedMapper siteFeedMapper, int maxLength, long delay) {
+        this.studioConfiguration = studioConfiguration;
+        this.auditServiceInternal = auditServiceInternal;
+        this.textEncryptor = textEncryptor;
+        this.siteFeedMapper = siteFeedMapper;
+        this.maxLength = maxLength;
+        this.delay = delay;
+    }
 
     @Override
-    public String encrypt(final String text) throws ServiceLayerException {
+    public String encrypt(final String siteId, final String text) throws ServiceLayerException {
         if (StringUtils.isEmpty(text) || text.length() > maxLength) {
             throw new InvalidParametersException("The provided text is invalid");
         }
         try {
             Thread.sleep(delay * 1000);
-            return textEncryptor.encrypt(text);
+            String encryptedToken = textEncryptor.encrypt(text);
+
+            var auth = SecurityContextHolder.getContext().getAuthentication();
+            String resolvedSiteId = StringUtils.isNotEmpty(siteId) ? siteId
+                    : studioConfiguration.getProperty(CONFIGURATION_GLOBAL_SYSTEM_SITE);
+            String targetId = UUID.randomUUID().toString();
```

Comment:
Can we store an MD5 sum of the encrypted token? That will let us connect the final token to the user and the audit record. You can always MD5 sum the token and match it to the audit record.

Reply:
Updated to use MD5 sum